src/sagemaker/estimator.py

```diff
@@ -695,26 +695,41 @@ def _stage_user_code_in_s3(self) -> str:
 
         Returns: S3 URI
         """
-        local_mode = self.output_path.startswith("file://")
-
-        if self.code_location is None and local_mode:
+        if is_pipeline_variable(self.output_path):
             code_bucket = self.sagemaker_session.default_bucket()
-            code_s3_prefix = "{}/{}".format(self._current_job_name, "source")
-            kms_key = None
-        elif self.code_location is None:
-            code_bucket, _ = parse_s3_url(self.output_path)
-            code_s3_prefix = "{}/{}".format(self._current_job_name, "source")
-            kms_key = self.output_kms_key
-        elif local_mode:
-            code_bucket, key_prefix = parse_s3_url(self.code_location)
-            code_s3_prefix = "/".join(filter(None, [key_prefix, self._current_job_name, "source"]))
-            kms_key = None
+            if self.code_location is None:
+                code_s3_prefix = "{}/{}".format(self._current_job_name, "source")
+                kms_key = self.output_kms_key
+            else:
+                code_bucket, key_prefix = parse_s3_url(self.code_location)
+                code_s3_prefix = "/".join(
+                    filter(None, [key_prefix, self._current_job_name, "source"])
+                )
+                kms_key = (
+                    self.output_kms_key
+                    if code_bucket == self.sagemaker_session.default_bucket()
+                    else None
+                )
         else:
-            code_bucket, key_prefix = parse_s3_url(self.code_location)
-            code_s3_prefix = "/".join(filter(None, [key_prefix, self._current_job_name, "source"]))
+            local_mode = self.output_path.startswith("file://")
+            if self.code_location is None:
+                code_bucket, _ = parse_s3_url(self.output_path)
+                code_s3_prefix = "{}/{}".format(self._current_job_name, "source")
+                kms_key = None if local_mode else self.output_kms_key
+            else:
+                code_bucket, key_prefix = parse_s3_url(self.code_location)
+                code_s3_prefix = "/".join(
+                    filter(None, [key_prefix, self._current_job_name, "source"])
+                )
 
-            output_bucket, _ = parse_s3_url(self.output_path)
-            kms_key = self.output_kms_key if code_bucket == output_bucket else None
+                output_bucket, _ = parse_s3_url(self.output_path)
+                kms_key = (
+                    None
+                    if local_mode
+                    else self.output_kms_key
+                    if code_bucket == output_bucket
+                    else None
+                )
```

Comment:
Nit:  it's a bit hard to follow. Can we add a () so it looks like:
```
kms_key = (
                    None
                    if local_mode
                    else (self.output_kms_key
                    if code_bucket == output_bucket
                    else None)
                )
```
Or expend this to 
```
if local_mode:
 kms_key = ...
elif code_bucket == output_bucket:
 kms_key = ...
else:
 kms_key = ...
```

Reply:
this is failing local_mode, got a new revision.